src/FsCheck/StateMachine.fs

```diff
@@ -118,10 +118,12 @@ type Machine<'Actual,'Model>(maxNumberOfCommands:int) =
     abstract Next : 'Model -> Gen<Operation<'Actual,'Model>>
 
 [<StructuredFormatDisplayAttribute("{StructuredToString}")>]
-type MachineRun<'Actual, 'Model> =
-    { Setup : 'Model * Setup<'Actual,'Model> 
-      Operations : list<Operation<'Actual,'Model> * 'Model>
-      TearDown : TearDown<'Actual> } with
+type MachineRun<'Actual, 'Model>(setup, operations, tearDown) =
+    member val internal Setup : 'Model * Setup<'Actual,'Model> = setup with get, set
+    member val internal Operations : list<Operation<'Actual,'Model> * 'Model> = operations with get, set
+    member val internal TearDown : TearDown<'Actual> = tearDown with get, set
+    member val internal UsedSize = -1 with get, set
+    member this.Clone() = this.MemberwiseClone() :?> MachineRun<'Actual, 'Model>
```

Comment:
Ho there! You made that beautiful immutable record type into a ugly mutable type? There better be some good explanation for this :)